src/components/Game/GameSmartPiano.tsx

```diff
@@ -50,7 +54,7 @@ const GameSmartPiano: React.FC<Props> = ({
 function areEqual(prevProps: Props, nextProps: Props) {
   return (
     prevProps.keyWidth === nextProps.keyWidth &&
-    prevProps.currentMapping === nextProps.currentMapping
+    prevProps.indexToNotesMap === nextProps.indexToNotesMap
```

Comment:
Since we do not expect it to change during the game (I guess), you can remove this, perhaps this is the reason for the rerendering bc i think this always return false because its a new array each time?

Reply:
instead, check for startTime here, bc initially the start time is negative i think, when the useEffect in the GameViews, then the proper start time will be set

Reply:
Okay. (I didn't dare to meddle with this just in case I screw up my part.) Thanks for the reminder!